zeth-contracts/contracts/BaseMixer.sol

```diff
@@ -58,19 +58,43 @@ contract BaseMixer is MerkleTreeMiMC7, ERC223ReceivingContract {
     // Here we use 2 inputs and 2 outputs (it is a 2-2 JS)
     // The number or public inputs is: 1 (the root) + 2 for each digest (nullifiers, commitments) + (1 + 1) (in and out public values) field elements
     uint constant jsIn = 2; // Nb of nullifiers
-    uint constant jsOut = 2; // Nb of commitments/ciphertexts
+    uint constant jsOut = 2; // Nb of commitments/cipher-texts
 
-    // We have 2 field elements for each digest (nullifierS (jsIn), commitmentS (jsOut), h_iS (jsIn) and h_sig)
-    // The root, v_pub_in and v_pub_out are all represented by one field element, so we have 1 + 1 + 1 extra public values
-    uint constant nbInputs = 1 + 2 * (jsIn + jsOut) + 1 + 1 + 2 * (1 + jsIn);
+    // Size of the public values in bits
+    uint constant size_value = 64;
+
+    // Constants regarding the hash digest length, the prime number used and its associated length in bits and the max values (v_in and v_out)
+    // uint r = 21888242871839275222246405745257275088548364400416034343698204186575808495617;
+    // field_capacity = ceil ( log_2(r) ) - 1
+    uint constant digest_length = 256;
+    uint constant field_capacity = 253;
+
+    // Variable representing the number of "residual" bits we can expect from converting a hash digest into a field element
+    // see primary input `residual_bits` in Reminder below
+    uint packing_residue_length = digest_length > field_capacity ? digest_length % field_capacity : 0;
+
+    // Total number of residual bits from packing of 256-bit long string into 253-bit long field elements
+    // to which are added the public value of size 64 bits
+    uint length_bit_residual = 2 * size_value + packing_residue_length * (1 + 2 * jsIn + jsOut);
+    // Number of field elements needed to pack this number of bits (the operation (a+b-1)/b represents div_ceil(a,b))
+    uint nb_field_residual = (length_bit_residual + field_capacity - 1) / field_capacity;
+
+    // The number of public inputs is:
+    // - 1 (the root)
+    // - jsIn (the nullifiers)
+    // - jsOut (the commitments)
+    // - 1 (hsig)
+    // - JsIn (the message auth. tags)
+    // - nb_field_residual (the residual bits not fitting in a single field element and the in and out public values)
+    uint nbInputs = 1 + jsIn + jsOut + 1 + jsIn + nb_field_residual;
 
     // Contract variable that indicates the address of the token contract
     // If token = address(0) then the mixer works with ether
     address public token;
 
     // The unit used for public values (ether in and out), in Wei. Must match
     // the python wrappers. Use Szabos (10^12 Wei).
-    uint64 constant public_unit_value_wei = 1 szabo;
+    uint constant public_unit_value_wei = 1 szabo;
```

Comment:
Was there some reason to change the type here?

Reply:
No reason! 
It must be due to a rebase and I missed it, changing it now.

Reply:
@dtebbs corrected.
I had to update  `vpub_in_zeth_units;` to uint however to prevent an integer overflow (we discussed about this in november on slack).